localstack-core/localstack/services/sqs/provider.py

```diff
@@ -621,29 +651,37 @@ class SqsDeveloperEndpoints:
 
     def __init__(self, stores=None):
         self.stores = stores or sqs_stores
-        self.service = load_service("sqs-query")
-        self.serializer = create_serializer(self.service)
 
     @route("/_aws/sqs/messages", methods=["GET", "POST"])
-    @aws_response_serializer("sqs-query", "ReceiveMessage")
+    @sqs_auto_protocol_aws_response_serializer("sqs", "ReceiveMessage")
     def list_messages(self, request: Request) -> ReceiveMessageResult:
         """
         This endpoint expects a ``QueueUrl`` request parameter (either as query arg or form parameter), similar to
         the ``ReceiveMessage`` operation. It will parse the Queue URL generated by one of the SQS endpoint strategies.
         """
-        # TODO migrate this endpoint to JSON (the new default protocol for SQS), or implement content negotiation
-        if "Action" in request.values and request.values["Action"] != "ReceiveMessage":
-            raise CommonServiceException(
-                "InvalidRequest", "This endpoint only accepts ReceiveMessage calls"
-            )
 
-        if not request.values.get("QueueUrl"):
+        if "x-amz-" in request.mimetype or "x-www-form-urlencoded" in request.mimetype:
+            # only parse the request using a parser if it comes from an AWS client
+            protocol = get_sqs_protocol(request)
+            operation, service_request = create_parser(
+                load_service("sqs", protocol=protocol)
```

Comment:
nit: I've updated it, as we're moving away from the `sqs-query` service name to the "service name" + "protocol" way of creating/selecting serializer and parser.